### ISSUE TITLE
RangeError for invalid codepoints

### ISSUE THREAD

**Issue by tom-lord** [jaynetics/regexp_property_values#1]
Codepoints in the range: `0xD800 - 0xDFFF` are not valid in UFF-8. This means certain properties cannot be listed:

```
RegexpPropertyValues['age=8.0'].matched_characters
#=> RangeError: invalid codepoint 0xD800 in UTF-8
```

How can a list of matched characters be generated?

The affected properties are:

```
age=*
surrogate
inhighsurrogates
inlowsurrogates
inhighprivateusesurrogates
any
assigned
unknown
other
```

**Comment by jaynetics** [comment]
hi!

the quick workaround would be

```ruby
RegexpPropertyValues[some_property]
  .matched_codepoints
  .filter_map { |cp| cp.chr('utf-8') unless cp.between?(0xD800, 0xDFFF) }
```

i guess `regexp_property_values` could do this filtering itself.

however, `#matched_characters` is quite slow (~0.3sec) and wasteful for large properties, as it does build a 1-character string for each character, and there can be more than a million of them in a property.

if you want to use this for `regexp-examples`, you'd get much better performance with something like

```ruby
RegexpPropertyValues[some_property]
  .matched_codepoints
  .shuffle
  .each_with_object([]) do |cp, acc| 
    acc << cp.chr('utf-8') unless cp.between?(0xD800, 0xDFFF)
    break if acc.size == desired_sample_size
  end
```

**Comment by jaynetics** [comment]
fixed in v1.5.2 (which omits surrogates from `#matched_characters`)
